server/sonar-server/src/test/java/org/sonar/server/ui/ws/ComponentActionTest.java

```diff
@@ -219,6 +219,50 @@ public void return_component_info_when_snapshot() throws Exception {
   }
 
   @Test
+  public void return_component_info_when_file_on_master() throws Exception {
+    init();
+    OrganizationDto organization = dbTester.organizations().insertForKey("my-org2");
+    ComponentDto main = componentDbTester.insertMainBranch(organization, p -> p.setName("Sample"), p -> p.setDbKey("sample"));
+    userSession.addProjectPermission(UserRole.USER, main);
+
+    ComponentDto dirDto = newDirectory(main, "src");
+    componentDbTester.insertComponent(dirDto);
```

Comment:
You can inline dirDto

Reply:
I certainly can. But will it be a good idea? That would mean replacing 3 lines with 1 longer line:

    ComponentDto fileDto = newFileDto(branch, componentDbTester.insertComponent(newDirectory(branch, "src")))

Reply:
Sorry, it was not clear, I was more thinking of : 
`ComponentDto dirDto = componentDbTester.insertComponent(newDirectory(main, "src"))`

Reply:
Oh I see, I agree with that. I think "inline" means "replace with implementation", and I interpreted that as eliminating the local variables completely.